### PR TITLE
Port ACT-4304 to Activiti 7

### PR DESCRIPTION
**Pull request by daisuke-yoshimoto**
I ported the following fix to Activiti 7(develop branch).
https://github.com/Activiti/Activiti/commit/29e3052f1a5afb2867a3cbee0910167e4f8c6806
https://community.alfresco.com/thread/231660-getting-error-on-transient-variable-usage-in-activiti

### CODE REVIEW

**Comment by daisuke-yoshimoto** [comment]
@Salaboy @balsarori 
There is one concern.
Is there any other fix that port is leaked out from the 6.x branch?

**Comment by daisuke-yoshimoto** [comment]
@balsarori 
Especially it is not in a hurry, but it is an easy fix, so can you check it?

File: activiti-engine/src/test/java/org/activiti/engine/test/api/variables/TransientVariablesTest.java [hunk 1]
```diff
@@ -51,6 +51,12 @@ public void testSetTransientVariableInServiceTask() {
     assertNull(runtimeService.getVariable(processInstance.getId(), "response"));
   }
   
+  // Test case for ACT-4304
+  @Deployment
+  public void testTransientVariableDeletion() {
+    runtimeService.startProcessInstanceByKey("transientVarsTest");
```
Comment (balsarori) [inline activiti-engine/src/test/java/org/activiti/engine/test/api/variables/TransientVariablesTest.java]: adding an assert here might stop Codacy from complaining

Comment (daisuke-yoshimoto) [reply activiti-engine/src/test/java/org/activiti/engine/test/api/variables/TransientVariablesTest.java]: @balsarori 
I fixed it. Please review again.
Also, should I backport this fix to 6.x branch that you first fixed?

Comment (balsarori) [reply activiti-engine/src/test/java/org/activiti/engine/test/api/variables/TransientVariablesTest.java]: @daisuke-yoshimoto yes please will be great to port it to 6.x

**Comment by daisuke-yoshimoto** [comment]
@Salaboy @erdemedeiros @ryandawsonuk 
Is it OK to merge to this pull request?

**Comment by erdemedeiros** [comment]
@daisuke-yoshimoto merged. Sorry for the delay.

**Comment by daisuke-yoshimoto** [comment]
@erdemedeiros Thanks.

**Pull request merged**